src/pipecat/serializers/exotel.py

```diff
@@ -0,0 +1,162 @@
+#
+# Copyright (c) 2024–2025, Daily
+#
+# SPDX-License-Identifier: BSD 2-Clause License
+#
+
+import base64
+import json
+from typing import Optional
+
+import loguru
+from pydantic import BaseModel
+
+from pipecat.audio import utils
+from pipecat.frames.frames import (
+    AudioRawFrame,
+    Frame,
+    InputAudioRawFrame,
+    InputDTMFFrame,
+    KeypadEntry,
+    StartFrame,
+    StartInterruptionFrame,
+    TransportMessageFrame,
+    TransportMessageUrgentFrame,
+)
+from pipecat.serializers.base_serializer import FrameSerializer, FrameSerializerType
+
+
+class ExotelFrameSerializer(FrameSerializer):
+    """Serializer for Exotel Media Streams WebSocket protocol.
+
+    This serializer handles converting between Pipecat frames and Exotel's WebSocket
+    media streams protocol. It supports audio conversion, DTMF events, and automatic
+    call termination.
+    """
+
+    class InputParams(BaseModel):
+        """Configuration parameters for ExotelFrameSerializer.
+
+        Attributes:
+            exotel_sample_rate: Sample rate used by Exotel, defaults to 8000 Hz.
+            sample_rate: Optional override for pipeline input sample rate.
+        """
+
+        exotel_sample_rate: int = 8000
+        sample_rate: Optional[int] = None
+
+    def __init__(
+        self, stream_sid: str, call_sid: Optional[str] = None, params: Optional[InputParams] = None
+    ):
+        """Initialize the ExotelFrameSerializer.
+
+        Args:
+            stream_sid: The Exotel Media Stream SID.
+            call_sid: The associated Exotel Call SID (optional, not used in this implementation).
+            params: Configuration parameters.
+        """
+        self._stream_sid = stream_sid
+        self._params = params or ExotelFrameSerializer.InputParams()
+
+        self._exotel_sample_rate = self._params.exotel_sample_rate
+        self._sample_rate = 0  # Pipeline input rate
+
+        self._resampler = utils.create_default_resampler()
```

Comment:
Let's import:
```
from pipecat.audio.utils import create_default_resampler, pcm_to_ulaw, ulaw_to_pcm
```
and just use `create_default_resampler` directly: 
```suggestion
        self._resampler = create_default_resampler()
```

Reply:
updated